pkgs/development/python-modules/spacy/default.nix

```diff
@@ -137,12 +142,14 @@ buildPythonPackage rec {
     tests.annotation = callPackage ./annotation-test { };
   };
 
+  __darwinAllowLocalNetworking = true; # needed for test_find_available_port
+
   meta = {
     description = "Industrial-strength Natural Language Processing (NLP)";
     homepage = "https://github.com/explosion/spaCy";
     changelog = "https://github.com/explosion/spaCy/releases/tag/release-v${version}";
     license = lib.licenses.mit;
-    maintainers = [ ];
+    maintainers = [ lib.maintainers.sarahec ];
```

Comment:
https://github.com/NixOS/nixpkgs/pull/428974#pullrequestreview-3061229006

Reply:
Any changes for me?

Reply:
Perhaps using `maintainers = with lib.maintainers; [ sarahec ];` but I have no strong opinion about it...

Reply:
I wonder how many reviewers understand why `with lib;` is deprecated but `with lib.maintainers;` isn't. (My understanding is it has to do the size of `lib` and the impact of including the whole thing.)

I've definitely had reviewers who insist that both `with` statements be removed.

Reply:
It’s true that this can be quite confusing.

Personally, I tend to discourage the use of `with lib;` especially when used in front of attribute sets. The reason is that it brings the entire `lib` namespace into scope, which mixes a large number of attributes with local variables. This creates ambiguity both for contributors and for tooling like IDEs and LSPs, making the code harder to navigatee, specially for newcomers.

That said, I’m more inclined to use with in front of lists than attribute sets. The use of with `lib.maintainers;` is generally accepted in `nixpkgs` because it’s simple, well-scoped, and its contents (maintainer usernames) are just leaf values, so the risk of confusion is much lower.